lib/Sema/DerivedConformanceCodingKey.cpp

```diff
@@ -405,8 +405,9 @@ static bool canSynthesizeCodingKey(TypeChecker &tc, Decl *parentDecl,
       return false;
   }
 
-  if (!enumDecl->getInherited().empty() &&
-      enumDecl->getInherited().front().isError())
+  auto inherited = enumDecl->getInherited();
+  if (!inherited.empty() && inherited.front().wasValidated() &&
```

Comment:
To give some context — this check was copied in from `DerivedConformanceRawRepresentable`, which performs a similar check. At the point of synthesizing conformance, AFAIK, we've already validated that a `RawRepresentable` enum has a `RawValue` of one of the types that's representable by a literal (i.e. one of the stdlib types). Since the enum already has a `RawValue` type which comes from the stdlib, is it possible for the type to not validate? All of the stdlib types get validated at one point or another, so I don't really see the point in performing this check for `RawRepresentable`.

Same here — the only place we fall into here is if the `RawValue` type is already either `String` or `Int`, which will get validated one way or another. Is there a benefit to performing eager validation here (or alternatively, even checking for validation in the first place)?